### ISSUE TITLE
Python/PyTorch Version Inquiry

### ISSUE THREAD

**Issue by zezhishao** [notaryanramani/stable-difussion#1]
Hello, thank you for contributing this open source project!

I would like to know which versions of Python and PyTorch were used to build this project, as I encountered some bugs when running it.

Looking forward to your reply!

**Comment by notaryanramani** [comment]
Hi,

If I remember correctly, I used Python 3.9 and PyTorch with Cuda 12.1. 

Would you like to share the error stack to help me debug the issue?

**Comment by zezhishao** [comment]
Hi there! Thank you so much for your quick response.

This repository has been extremely helpful for my understanding of Stable Diffusion, and I’m more than happy to contribute improvements to the code.

I’ve made a few modifications so the code can run again. However, since I’m not a researcher in the field of computer vision, I’m not entirely sure if everything I changed is correct. You can find my changes in the PR. 

My environment details are listed below:
```python
Python: 3.11
PyTorch: 2.6.0+cu124
Transformers: 4.40.1
torchvision: 0.21.0+cu124
```


Here’s a summary of what I’ve modified:

---

**`main.py`:**
1. I couldn’t find the annotation file for `test2017` on the official COCO website. (I’m new to this area, so please pardon any naive mistakes.) So I changed it to `val2017`.
2. `DiffusionModel` does not have a `train` function; `train_model` appears to be the correct one.
3. I manually passed in the `batch_size` to avoid potential bugs from modifying the batch size in the config.

---

**`stable_diffusion/bert.py`:**
1. The `tokens` generated by the BERT tokenizer don’t seem to be placed on the correct device yet.

---

**`stable_diffusion/dm.py`:**
1. I unified the `batch_size` variable in the `train_model` function to be a function parameter.
2. It appears that `self` (i.e., `DiffusionModel`) does not have a `device` attribute or an `m` function.

---

**`stable_diffusion/scheduler.py`:**
1. PyTorch does not support slicing like NumPy’s `[::-1]`.

---

**`stable_diffusion/unet.py`:**
1. Some dimensions don’t seem to match up. I made a few changes, but ***I’m not entirely sure they are correct.***
2. The `UNet` class is missing a `super().__init__()` call, which leads to errors.
3. The `res` and `attn` blocks are missing parameters in their forward pass.

---

**`stable_diffusion/utils.py`:**
1. The positional embedding (`pe`) in `TimePositionEmbedding` is not placed on the correct device.

---

Thank you again for providing this code. I look forward to hearing your feedback and would love to continue contributing to this project!

**Comment by notaryanramani** [comment]
That makes sense! I encountered these bugs while developing the project.

I developed this project on my Desktop back at home, maybe I forgot to commit/push the latest changes. I unfortunately did not have access to the desktop anymore because I am in another country, so I will have a look at the pull request.

Thank you so much!
